packages/typespec-azure-resource-manager/src/private.decorators.ts

```diff
@@ -48,6 +48,31 @@ export function $omitIfEmpty(context: DecoratorContext, entity: Model, propertyN
   }
 }
 
+export function $enforceConstraint(
+  context: DecoratorContext,
+  entity: Operation | Model,
+  sourceType: Model,
+  constraintType: Model
+) {
+  if (sourceType !== undefined && constraintType !== undefined) {
+    // walk the baseModel chain until find a match or fail
+    let baseType: Model | undefined = sourceType;
+    do {
+      if (baseType === constraintType) return;
```

Comment:
Do we want to do this, though?  It feels like this will be too restrictive for brownfield APIs.    There are brownfield APIs that just mark their own internal resource as x-ms-azure-resource, and these will have to depend on Resource now.

Reply:
I thought about it. But given that Foundations.Resource is really the lowest basic model, this gives us an opportunity not letting them continue on the forked path. I don't see there is any reason not to extend from this.